libs/template/presets/src/features/fulfillment/components/fulfillment-root/fulfillment-root.component.ts

```diff
@@ -11,6 +16,35 @@ export class FulfillmentRootComponent extends LitElement {
 
   router = new LitRouter(this, this.extraRoutes ?? []);
 
+  constructor(
+    protected authService = resolve(OauthService),
+    protected routerService = resolve(RouterService)
+  ) {
+    super();
+  }
+
+  @subscribe()
+  protected guard$ = combineLatest([
+    this.authService.isAuthenticated(),
+    this.routerService.currentRoute(),
+  ]).pipe(
+    tap(([authenticated, currentRoute]) => {
+      if (
+        !authenticated &&
+        currentRoute !== '/login' &&
+        !currentRoute.startsWith('/oauth')
+      ) {
+        this.authService.login();
+      }
+      if (
+        authenticated &&
+        (currentRoute.startsWith('/login') || currentRoute.startsWith('/oauth'))
+      ) {
+        this.routerService.navigate('/');
+      }
+    })
+  );
```

Comment:
I think it's better to move this to a separate plugin in the `application` package instead of embedding it into a component.